trie/snapshotTrieStorageManager_test.go

```diff
@@ -31,55 +34,117 @@ func TestNewSnapshotTrieStorageManager(t *testing.T) {
 	assert.False(t, check.IfNil(stsm))
 }
 
-func TestNewSnapshotTrieStorageManager_GetFromOldEpochsWithoutCache(t *testing.T) {
+func TestNewSnapshotTrieStorageManager_Get(t *testing.T) {
 	t.Parallel()
 
-	_, trieStorage := newEmptyTrie()
-	getFromOldEpochsWithoutCacheCalled := false
-	trieStorage.mainStorer = &trie.SnapshotPruningStorerStub{
-		GetFromOldEpochsWithoutAddingToCacheCalled: func(_ []byte) ([]byte, core.OptionalUint32, error) {
-			getFromOldEpochsWithoutCacheCalled = true
-			return nil, core.OptionalUint32{}, nil
-		},
-	}
-	stsm, _ := newSnapshotTrieStorageManager(trieStorage, 0)
-
-	_, _ = stsm.Get([]byte("key"))
-	assert.True(t, getFromOldEpochsWithoutCacheCalled)
+	t.Run("closed storage manager should error", func(t *testing.T) {
+		t.Parallel()
+
+		_, trieStorage := newEmptyTrie()
+		trieStorage.mainStorer = &trie.SnapshotPruningStorerStub{}
+		stsm, _ := newSnapshotTrieStorageManager(trieStorage, 0)
+		_ = stsm.Close()
+
+		val, err := stsm.Get([]byte("key"))
+		assert.Equal(t, errorsMx.ErrContextClosing, err)
+		assert.Nil(t, val)
+	})
+	t.Run("GetFromOldEpochsWithoutAddingToCache returns db closed should error", func(t *testing.T) {
+		t.Parallel()
+
+		_, trieStorage := newEmptyTrie()
+		trieStorage.mainStorer = &trie.SnapshotPruningStorerStub{
+			GetFromOldEpochsWithoutAddingToCacheCalled: func(_ []byte) ([]byte, core.OptionalUint32, error) {
+				return nil, core.OptionalUint32{}, storage.ErrDBIsClosed
+			},
+		}
+		stsm, _ := newSnapshotTrieStorageManager(trieStorage, 0)
+
+		val, err := stsm.Get([]byte("key"))
+		assert.Equal(t, storage.ErrDBIsClosed, err)
+		assert.Nil(t, val)
+	})
+	t.Run("should work from old epochs without cache", func(t *testing.T) {
+		t.Parallel()
+
+		_, trieStorage := newEmptyTrie()
+		getFromOldEpochsWithoutCacheCalled := false
+		trieStorage.mainStorer = &trie.SnapshotPruningStorerStub{
+			GetFromOldEpochsWithoutAddingToCacheCalled: func(_ []byte) ([]byte, core.OptionalUint32, error) {
+				getFromOldEpochsWithoutCacheCalled = true
+				return nil, core.OptionalUint32{}, nil
+			},
+		}
+		stsm, _ := newSnapshotTrieStorageManager(trieStorage, 0)
+
+		_, _ = stsm.Get([]byte("key"))
+		assert.True(t, getFromOldEpochsWithoutCacheCalled)
+	})
 }
 
-func TestNewSnapshotTrieStorageManager_PutWithoutCache(t *testing.T) {
+func TestNewSnapshotTrieStorageManager_Put(t *testing.T) {
```

Comment:
Might rename from `TestNewSnapshotTrieStorageManager*` to `TestSnapshotTrieStorageManager*`

Reply:
for the entire file